src/color.rs

```diff
@@ -1072,3 +1245,45 @@ where
 
     Ok(into_color(lightness, chroma, hue, alpha))
 }
+
+#[inline]
+fn parse_color_function<'i, 't, ComponentParser>(
+    component_parser: &ComponentParser,
+    arguments: &mut Parser<'i, 't>,
+) -> Result<Color, ParseError<'i, ComponentParser::Error>>
+where
+    ComponentParser: ColorComponentParser<'i>,
+{
+    let location = arguments.current_source_location();
+    let color_space = arguments.try_parse(|i| {
+        let ident = i.expect_ident()?;
+        PredefinedColorSpace::from_str(ident)
+            .map_err(|_| location.new_unexpected_token_error(Token::Ident(ident.clone())))
+    })?;
+
+    macro_rules! parse_component {
+        () => {{
+            if let Ok(c) = arguments.try_parse(|i| component_parser.parse_number_or_percentage(i)) {
```

Comment:
Why is this optional? This is allowing to parse `color(srgb)` which seems wrong per spec. We eventually need to support `none`, but that's a different issue.

Reply:
I believe this should just do:

```
let c1 = component_parser.parse_number_or_percentage(arguments)?;
...
```

unless I'm missing something?